### DISCUSSION TITLE
no cookie is set up on change locale, how to preserve locale across browser open/close sessions ???

### DISCUSSION THREAD

**Discussion by kristijorgji** [i18next/next-i18next#1418] [Q&A] [answered]
I have a function that changes locale on button click, like this:

```
    const changeLanguage = (language: string) => {
        i18n?.changeLanguage(language, () => {
            p.router.push(
                getRouteSource({
                    pathname: p.router.pathname,
                    query: p.router.query,
                }),
                undefined,
                { locale: language }
            );
        });
    };
```

Changing the locale works great, but in version 6 (now upgraded to last 8.x), i saw also some cookie being set up like next_language, which on next time user came into the site found the site in the last used language.

I don't see this cookie set up anymore.

I get scenarios like this now:
1. i go to domain.com (default en)
2. i choose locale de, switch correctly to domain.com/de and all routing is done in de after some hacky solution i did for localized urls
3. i close browser, then open domain.com again. I expect now to see domain.com/de_DE (as my last used locale was de_DE)

Is there a way to store cookie with library ? or should i implement some custom solution to this

Thanks

**Comment by isaachinman** [comment, marked as answer]
NextJs handles locale and routing now – `next-i18next` no longer handles this behaviour whatsoever.

**Reply by kristijorgji** [reply]
allright thanks, will look into how to change this in next itself. changeLanguage method of next-i18next is working fine in client side, just need to find a way to "persist" this user preference, until they themselves change locale again

**Reply by isaachinman** [reply]
You should _not_ be changing the language via i18next. You should change it via NextJs itself.

**Reply by kristijorgji** [reply]
how can i do that via next itself ? i cannot find anything in their documentation..

**Reply by isaachinman** [reply]
Please post on StackOverflow for help in the future. GitHub is not the place to ask for help in reading documentation.

https://nextjs.org/docs/advanced-features/i18n-routing#transition-between-locales
